File: BOJ/CLASS 2/Python/1018.py
# 체스판 다시 칠하기
n, m = map(int, input().split())
data = [input() for _ in range(n)]

white = ['WBWBWBWB', 'BWBWBWBW', 'WBWBWBWB', 'BWBWBWBW', 'WBWBWBWB', 'BWBWBWBW', 'WBWBWBWB', 'BWBWBWBW']
black = ['BWBWBWBW', 'WBWBWBWB', 'BWBWBWBW', 'WBWBWBWB', 'BWBWBWBW', 'WBWBWBWB', 'BWBWBWBW', 'WBWBWBWB']

result = 2500

# 8x8 크기의 체스판 자르기
for i in range(n - 7):
    for j in range(m - 7):
        count_w = 0
        count_b = 0

        # 다시 칠해야 하는 정사각형의 개수 구하기
        for k in range(8):
            for s in range(8):
                if data[i + k][j + s] != white[k][s]: count_w += 1
                if data[i + k][j + s] != black[k][s]: count_b += 1

        result = min(result, count_w, count_b)

print(result)

File: BOJ/CLASS 2/JavaScript/1018.js
// 체스판 다시 칠하기

const fs = require('fs');
const input = fs.readFileSync('예제.txt').toString().trim().split('\n');

let white = [
  'WBWBWBWB',
  'BWBWBWBW',
  'WBWBWBWB',
  'BWBWBWBW',
  'WBWBWBWB',
  'BWBWBWBW',
  'WBWBWBWB',
  'BWBWBWBW',
];

let black = [
  'BWBWBWBW',
  'WBWBWBWB',
  'BWBWBWBW',
  'WBWBWBWB',
  'BWBWBWBW',
  'WBWBWBWB',
  'BWBWBWBW',
  'WBWBWBWB',
];

function solution(input) {
  [size, ...arr] = input;
  [row, col] = size.split(' ').map((x) => Number(x));
  arr = arr.map((str) => str.trim('\r').split(''));
  let answer = [];

  // 8 * 8 크기의 체스판을 만드는 경우
  for (let i = 0; i <= row - 8; i++) {
    for (let j = 0; j <= col - 8; j++) {
      // whiteCheck, blackCheck
      let cnt1 = (cnt2 = 0);

      for (let k = 0; k < 8; k++) {
        for (let s = 0; s < 8; s++) {
          if (white[k][s] !== arr[i + k][j + s]) cnt1++;
          if (black[k][s] !== arr[i + k][j + s]) cnt2++;
        }
      }
      answer.push(cnt1);
      answer.push(cnt2);
    }
  }

  return Math.min(...answer);
}

console.log(solution(input));
